# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

Lane 2, Refresh Opportunity Scoring.

I chose this lane because it has a clear task which I can feasibly investigate during this internship. The task is to help editors decide which pages to review out of a large number of pages, not all of which may require an update. Also, this lane will allow me to explore how multiple signals can be combined to produce a ranking model. Finally, working on this lane will challenge my technical skills, while allowing me to contribute towards a real-world problem

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

My work improves the decision of which pages to review  first for potential updates.

The output my work produces can be observed by editors who may use it to decide which pages to review.

The costs of wrong recommendations are:

    * False positive: The editor potentially wastes time and resources reviewing a page that does not need to be reviewed. This could reduce their trust in the model.

    * False negative: A page with declining performance may go unnoticed, which means an opportunity to improve it may be missed

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [2]:
import os, sys, subprocess

IN_COLAB = "google.colab" in sys.modules
REPO_URL = "https://github.com/flyrank-bih/flyrank-ml-internship-starter"
REPO_DIR = "flyrank-ml-internship-starter"

if IN_COLAB:
    if not os.path.isdir(REPO_DIR):
        subprocess.run(["git", "clone", "--depth", "1", REPO_URL, REPO_DIR], check=True)
    os.chdir(REPO_DIR)
    subprocess.run([sys.executable, "-m", "pip", "install", "-q", "-r", "requirements.txt"], check=True)
elif os.path.basename(os.getcwd()) == "notebooks":
    os.chdir("..")

import pandas as pd, numpy as np
df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print(f"Number of pages: {df['content_id'].nunique()}\n")
display(df["engagement_rate"].describe())

display(df['trend_direction'].value_counts())

display(df["trend_pct"].describe())

Number of pages: 30000



,engagement_rate
count,30000.000000
mean,2.534520
std,8.310096
min,0.000000
25%,0.000000
50%,0.000000
75%,1.350000
max,100.000000


,count
trend_direction,
down,16262
stable,5962
up,4388
new,2236
flat,1152


,trend_pct
count,26612.000000
mean,-4.785969
std,473.861780
min,-100.000000
25%,-62.600000
50%,-33.500000
75%,0.000000
max,44900.000000


As shown in the above cell, there are 30,000 pages in the dataset, which means it would be time consuming and inefficient for editors to review every page, while a prioritisation method could help them focus on pages that need their attention. In addition, 16262 pages (54.2% of the dataset) have a downward trend, further indicating that some pages should be prioritised over others for review. In addition, there are a large number of variables in the dataset related to its performance, such as engagement rate, making a scalable static rule difficult to implement. Therefore, I will explore if a machine learning ranking model can prioritise which pages may need to be reviewed.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

My work will be able to make claims such as having observed patterns in the dataset that suggest declining performance. I can claim that my model will use these patterns to rank pages in a way that support the decisions of the editors who have to decide which pages to review.

However, my work will never make definitive claims such as my model's rankings being the correct rankings for the pages that need reviewing. The model will guide the editors by bringing pages that may require review to their attention, but the final decision as to which pages need reviewing most urgently will be theirs to make.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.